lib/matplotlib/backend_bases.py

```diff
@@ -104,16 +104,12 @@ def _safe_pyplot_import():
         current_framework = cbook._get_running_interactive_framework()
         if current_framework is None:
             raise  # No, something else went wrong, likely with the install...
-        backend_mapping = {
-            'qt': 'qtagg',
-            'gtk3': 'gtk3agg',
-            'gtk4': 'gtk4agg',
-            'wx': 'wxagg',
-            'tk': 'tkagg',
-            'macosx': 'macosx',
-            'headless': 'agg',
-        }
-        backend = backend_mapping[current_framework]
+
+        from matplotlib.backends.registry import backendRegistry
+        backend = backendRegistry.framework_to_backend(current_framework)
+        if backend is None:
+            raise KeyError(backend)
```

Comment:
This gives exactly the same error as prior to this PR. But probably it should be more informative.

Reply:
```suggestion
            raise KeyError(current_framework)
```
would reproduce the previous behavior. But I would go with
```suggestion
            raise RuntimeError(f"No suitable backend for the current GUI framework {current_framework!r}")
```